src/containers/ClassView.ts

```diff
@@ -39,7 +39,10 @@ function mapDispatchToProps(dispatch: Dispatch<IStoreState>) {
       planName: string,
       parameters: {},
     ) => {
-      dispatch(actions.catalog.provision(instanceName, namespace, className, planName, parameters));
+      await dispatch(
+        actions.catalog.provision(instanceName, namespace, className, planName, parameters),
```

Comment:
can we dispatch the getCatalog call inside of the provision action? That way we can keep this as a simple mapping.

Reply:
That'd definitely an option. But I'd personally lean away from that because I like to keep my actions relatively explicit; At the moment the actions do exactly as their name implies (ie. `provision` provisions something). If we add the `getCatalog()` call to it, it sort of becomes more 'magical'. WDYT?

Reply:
Fair enough and I agree that `provision` should not do that. In other cases I've created another action (e.g. `provisionAndGetCatalog`), but perhaps that is becoming messy and it might just be easier to control this from the container (i.e. let the container have the logic about the flow of the actions, and keep actions simpler).

I'll leave it up to you then, if you think the container should do this then we can merge as-is :)!